srml/babe/src/lib.rs

```diff
@@ -131,16 +140,13 @@ decl_storage! {
 		/// (like everything else on-chain) it is public. For example, it can be
 		/// used where a number is needed that cannot have been chosen by an
 		/// adversary, for purposes such as public-coin zero-knowledge proofs.
-		EpochRandomness get(epoch_randomness): [u8; VRF_OUTPUT_LENGTH];
-
-		/// The randomness under construction
-		UnderConstruction: [u8; VRF_OUTPUT_LENGTH];
+		pub Randomness get(randomness): [u8; RANDOMNESS_LENGTH];
```

Comment:
This (and the 2 below), appear in the metadata as `[u8; RANDOMNESS_LENGTH]` as opposed to `[u8; 32]`. The macro seems to extract the information as-is instead of interpreting it. (The last bit is even more black magic). Effectively this yields less than useful info for the middleware parsers.

Just indicating the issue, not sure what the best long-term solution here would be -

- make the code above more magic-y (not using the constants)
- making the macros even more magic-y ()

cc @bkchr

Reply:
For now I will just hardcode the values but keep the constants and add a comment to use them once this works. @bkchr mentioned this needs type reflection.

Reply:
That is great - that is in active dev, so not tomorrow, but on the way. It also means we don't spend effort on something that will be replaced anyway.